ksqldb-engine/src/main/java/io/confluent/ksql/engine/EngineExecutor.java

```diff
@@ -478,7 +478,7 @@ private PersistentQueryMetadata executePersistentQuery(
   }
 
   private String buildPlanSummary(final QueryId queryId, final ExecutionStep<?> plan) {
-    return new PlanSummary(queryId, config.getConfig(false), engineContext.getMetaStore())
+    return new PlanSummary(queryId, config.getConfig(true), engineContext.getMetaStore())
```

Comment:
Is there a reason we weren't applying config overrides here? Seems like a bug. This fix is needed in order to pass the feature flag into the GroupByParamsBuilder.

Reply:
off the top of my head, I don't think there's any good reason. We should make sure that it's not being used in the pull query path as this (calling `getConfig(true)`) is shockingly expensive at high throughput.

Reply:
This helper method is only called for persistent and push queries, not pull queries, so we should be in the clear. Good call on checking!